### PR TITLE
Server modifications

### PR DESCRIPTION
**Pull request by samliebow**
Changes in rough order of significance:
1. Switching from the old MongoDB database to the new PostgreSQL database
2. Adding Redis for caching
3. Using Node clustering for performance improvement when on a multi-core system
4. Adding New Relic for monitoring and an Artillery test config file

### CODE REVIEW

File: db/artillery.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - loop:
+        - get:
+            url: "/api/community/{{ $randomNumber(1,10000000) }}"
+        count: 100
+  - name: "Look at one of top 10000 projects"
+    flow:
+      - loop:
+        - get:
+            url: "/api/community/{{ $randomNumber(1,10000) }}"
+        count: 100
```
Comment (jaweesner) [inline db/artillery.yml:19]: one thing you could have done is use scenario weights, which allow you to specify relatively how often different scenarios should be run (weight of 3/4 run 75% of the time)

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 node_modules/
 .DS_Store
-client/dist/bundle.js
\ No newline at end of file
+client/dist/bundle.js
+newrelic_agent.log
```
Comment (jaweesner) [inline .gitignore:4]: you might want to put your NR key file in here just as a preventative measure

File: server/basic-server.js [hunk 3]
```diff
@@ -1,44 +1,74 @@
...
+          SELECT title, creator, backers, 
+            name, city, country, avatar, 
+            projects_backed AS "fundedProjects"
+          FROM quickstarter.projects 
+          INNER JOIN quickstarter.projects_backers 
+            ON quickstarter.projects.id = quickstarter.projects_backers.project
+          INNER JOIN quickstarter.users
+            ON quickstarter.projects_backers.backer = quickstarter.users.id
+          WHERE quickstarter.projects.id = $1::integer;`, [id]);
+        client.set(id, JSON.stringify(result), 'EX', 60);
```
Comment (jaweesner) [inline server/basic-server.js]: no need to set an expiration time, redis defaults to a LRU ejection method

**Review by jaweesner — COMMENTED** [review]
Overall, good, I would personally have broken up the basic server into server req and db handling files to help with modularity.

File: .gitignore (hunk 2 above)
Comment (samliebow) [reply .gitignore:4]: I've stored my key in the parent folder for this repo, so it won't ever be accidentally committed.

File: db/artillery.yml (hunk 1 above)
Comment (samliebow) [reply db/artillery.yml:19]: My goal for this was 50-50 between the two scenarios, so I left off scenario weights here; if I fine-tune, I'll incorporate them.

File: server/basic-server.js (hunk 3 above)
Comment (samliebow) [reply server/basic-server.js]: Thanks, I'll fix and look at fine-tuning the memory limit.

**Comment by samliebow** [comment]
@jaweesner : can you do a quick re-review with the updates before I merge?

**Comment by jaweesner** [comment]
looks good

**Pull request merged**